apps/system/js/bluetooth.js

```diff
@@ -188,6 +198,11 @@ var Bluetooth = {
     var req = bluetooth.getDefaultAdapter();
     req.onsuccess = (evt) => {
       this.defaultAdapter = req.result;
+      // resolve the adapter request
+      if (this._resolveAdapterPromise) {
+        this._resolveAdapterPromise(this.defaultAdapter);
+        this._resolveAdapterPromise = null;
```

Comment:
As above comment, you can just do `this._getAdapterPromiseResolve(this.defaultAdapter);` without clean up the resolve cache. Thanks.